taskmanager/models.py

```diff
@@ -448,21 +332,16 @@ def launch(self):
             schedule = int(self.scheduling.timestamp())
             # NOTE: spool at param requires bytes
             kwargs["at"] = str(schedule).encode()
-        try:
-            task_id = exec_command_task.spool(self, **kwargs)
-        except Exception as e:
-            # TODO: better handle exception, might cause unexpected behaviours
-            Report.objects.create(
-                task=self,
-                invocation_result=Report.RESULT_FAILED,
-                log=f"TASK SPOOL FAILED!!!\n{e}",
-            )
+
+        # Spool the execution of the command
+        task_id = exec_command_task.spool(self, **kwargs)
+        if task_id:
+            self.spooler_id = task_id.decode("utf-8")
         else:
-            try:
-                self.spooler_id = task_id.decode("utf-8")
-            except AttributeError:
-                # NOTE: this occurs when no uWSGI is loaded (e.g. during tests)
-                pass
+            # This probably means the uWSGI spooler is unavailable and
+            # the task executed synchronously (e.g. during tests).
+            pass
+
```

Comment:
I re-wrote this part and removed the try-except block. I think it does more harm than good. For example, makes debugging a nightmare. Anyway, if an exception is thrown there (and it shouldn't), it should be probably raised; exceptions happening inside a task are already caught and handled in `exec_command_task` function.